tests/test_wfp/test_wfp.py

```diff
@@ -0,0 +1,1451 @@
+
+        # mock_convert.assert_called_once_with(["InvalidCountry"], to="ISO3")
+
+
+def test_wfp_inflation_get_data_no_data_found(wfp_inflation_post_load):
+    """
+    Test that WFPInflation's `get_data` method returns an empty DataFrame and logs a warning when no data is found.
+    """
+    # Clear existing data
+    wfp_inflation_post_load._data = {"Headline inflation (YoY)": {}}
+
+    # Mock logger warning
+    with mock.patch("bblocks_data_importers.config.logger.warning") as mock_warning:
+        # Call method
+        result = wfp_inflation_post_load.get_data(
+            indicators="Headline inflation (YoY)", countries=["USA"]
+        )
+
+        # Assertions
+        pd.testing.assert_frame_equal(result, pd.DataFrame())
+        mock_warning.assert_any_call("No data found for the requested countries")
+        # mock_warning.assert_called_once_with("No data found for the requested countries")
+
+
+def test_wfp_inflation_get_data_load_countries_when_none(wfp_inflation_post_load):
+    """
+    Test that `get_data` calls `load_available_countries` when `_countries` is None.
+    """
+    # Set `_countries` to None
+    wfp_inflation_post_load._countries = None
+
+    # Mock `load_available_countries` to populate `_countries`
+    with mock.patch.object(
+        wfp_inflation_post_load,
+        "load_available_countries",
+        side_effect=lambda: setattr(
+            wfp_inflation_post_load,
+            "_countries",
+            {
+                "USA": {"entity_code": 1, "name": "United States"},
+                "CAN": {"entity_code": 2, "name": "Canada"},
+            },
+        ),
+    ) as mock_load:
+        # Call the method
+        result = wfp_inflation_post_load.get_data()
+
+        # Assert that `load_available_countries` was called
+        mock_load.assert_called_once()
+
+        # Assert that the result contains the combined data
+        expected = pd.concat(
+            [
+                wfp_inflation_post_load._data["Headline inflation (YoY)"]["USA"],
+                wfp_inflation_post_load._data["Headline inflation (YoY)"]["CAN"],
+            ],
+            ignore_index=True,
+        )
+        pd.testing.assert_frame_equal(result, expected)
+
+
+def test_wfp_inflation_get_data_countries_as_string(wfp_inflation_post_load):
+    """
+    Test that WFPInflation's `get_data` method converts a single country string into a list.
+    """
+    # Mock `load_data`
+    with mock.patch.object(wfp_inflation_post_load, "load_data") as mock_load_data:
+        # Call method with `countries` as a string
+        wfp_inflation_post_load.get_data(countries="USA")
+
+        # Ensure `load_data` is called with the country wrapped in a list
+        mock_load_data.assert_called_once_with(
+            indicator_name="Headline inflation (YoY)", iso3_codes=["USA"]
+        )
+
+
+def test_wfp_inflation_clear_cache(wfp_inflation_post_load):
```

Comment:
there should also be a test that checks that an extraction is run correctly after `clear_cache` is called